### ISSUE TITLE
Request for batch inference

### ISSUE THREAD

**Issue by wlin16** [flagshippioneering/bio2token#1]
Thank you for the amazing work! Well done!

I was wondering if it would be possible for the author to prepare a .py file that allows users to specify the path of multiple PDB files, enabling the model to process them in batch mode. Currently, the test_pdb.py script only supports processing a single PDB file at a time.

Thank you again for your help!

**Comment by OliviaViessmann** [comment]
Hi, the model runs in batch mode, take a look at `test.py` or adopt the `test_pdb.py` to read all pdbs in a directory and batch those.

**Comment by wlin16** [comment]
> Hi, the model runs in batch mode, take a look at `test.py` or adopt the `test_pdb.py` to read all pdbs in a directory and batch those.

Thank you so much for your response and for pointing me to the test.py script. I did take a look at it, but unfortunately, the input for test.py appears to require a parquet file. However, the repository doesn’t seem to provide such a file, and I am not entirely sure what the structure of this parquet file should be or how I should prepare it.

I did try to adapt test_pdb.py to process multiple PDB files, but I have encountered issues with the mask generation part, which has been quite challenging to resolve on my own.

Given these difficulties, I would be incredibly grateful if you could kindly assist by providing some guidance or a script that can help with batch processing. Your help would be invaluable to our project, and we truly appreciate any support you can offer.

Thank you once again for your time and understanding!

**Comment by Sisistern123** [comment]
Hi @wlin16 
Me and @BetelgeuseBugFixer are currently working with Bio2Token [in here](https://github.com/BetelgeuseBugFixer/MaPra2025) and thought this code snippet might resolve your issue!

```
from models.bio2token.data.utils.utils import pdb_2_dict, uniform_dataframe, compute_masks, pad_and_stack_batch, filter_batch

def batch_pdbs_for_bio2token(pdbs, device):
    batch = []
    # read to dicts
    dicts = [pdb_2_dict(pdb) for pdb in pdbs]
    for pdb_dict in dicts:
        structure, unknown_structure, residue_name, residue_ids, token_class, atom_names_reordered = uniform_dataframe(
            pdb_dict["seq"],
            pdb_dict["res_types"],
            pdb_dict["coords_groundtruth"],
            pdb_dict["atom_names"],
            pdb_dict["res_atom_start"],
            pdb_dict["res_atom_end"],
        )
        batch_item = {
            "structure": torch.tensor(structure).float(),
            "unknown_structure": torch.tensor(unknown_structure).bool(),
            "residue_ids": torch.tensor(residue_ids).long(),
            "token_class": torch.tensor(token_class).long(),
        }
        batch_item = {k: v[~batch_item["unknown_structure"]] for k, v in batch_item.items()}

        batch.append(batch_item)

    # taken from config
    sequences_to_pad = {
        "structure": 0,
        "unknown_structure": True,
        "residue_ids": -1,
        "token_class": PAD_CLASS,
        "eos_pad_mask": 1,
        "structure_known_all_atom_mask": 0,
        "bb_atom_known_structure_mask": 0,
        "sc_atom_known_structure_mask": 0,
        "cref_atom_known_structure_mask": 0,
    }
    batch = filter_batch(batch, sequences_to_pad.keys())
    batch = pad_and_stack_batch(
        batch,
        sequences_to_pad,
        1
    )

    batch = compute_masks(batch, structure_track=True)
    batch = {k: v.to(device) for k, v in batch.items()}
    return batch
```
